### PR TITLE
`Development`: Switch from icon library to explicit reference

### PR DESCRIPTION
**Pull request by filip-gregurevic**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context

Change the icon usage from the icon library approach to the explicit reference approach.
The explicit reference approach involves explicitly importing the icon definition from the npm package, assigning it to the component's property and then binding this property to the icon input of the fa-icon component.

This approach is more verbose than the icon library approach, it makes it much easier to figure out where or whether the given icon is used. For example, with Find usages feature of the IDE. Another benefit of this approach is that it will produce a compile-time error if an icon is missing.

Also, this helps to further reduce the bundle size as icons are only loaded when needed.
### Description

Changed all icon imports and usages.

### Steps for Testing

Prerequisites:
???

1. Log in to Artemis
2. Click trough all screens (or as much as possible) on Artemis which has icons (almost all of them have)
3. Check that the icons are displayed
4. Have the dev-tools open and check if an error is displayed in the console

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

File: src/test/javascript/spec/test.module.ts [hunk 1]
```diff
@@ -76,8 +74,6 @@ import { MockTranslateService } from './helpers/mocks/service/mock-translate.ser
 export class ArtemisTestModule {
     constructor(iconLibrary: FaIconLibrary, dpConfig: NgbDatepickerConfig, translateService: TranslateService) {
```
Comment (krusche) [inline src/test/javascript/spec/test.module.ts:75]: I guess you can remove `iconLibrary: FaIconLibrary` in the constructor as well

**Comment by Malyuk-A** [comment]
image
image
image
There are sadly issues right from the first couple pages I visited 😅

**Review by Malyuk-A — APPROVED** [review]
I tested the PR again on TS1 and found no issues. I had the chrome dev tools open at all times and clicked around through Artemis through very different pages and subsystems. I have not received any errors in the console.
The code seems reasonable. I looked through 50 files (the rest should be very similar)
Approve! 👍

**Review by willergeorg — DISMISSED** [review]
Tested on TS1. Worked smoothly and did not encounter any errors in the console while clicking through a bundle of pages (Assessment dashboard, course detail page, exercise detail page, participations page, scores page of exam and course, exams page of a course, exercise groups of an exam). Code seems to be straight forward to me.

**Review by Malyuk-A — APPROVED** [review]
Tested on TS2. Re-approve

**Pull request merged**